test/widgets/app_test.dart

```diff
@@ -51,4 +54,56 @@ void main() {
       ]);
     });
   });
+
+  group('choose account page', () {
+    Future<void> setupChooseAccountPage(WidgetTester tester, {
+      List<Account>? accounts,
+    }) async {
+      addTearDown(testBinding.reset);
+
+      for (final account in accounts ?? []) {
+        await testBinding.globalStore
+          .insertAccount(account.toCompanion(false));
+      }
+
+      await tester.pumpWidget(
+        const MaterialApp(
+          localizationsDelegates: ZulipLocalizations.localizationsDelegates,
+          supportedLocales: ZulipLocalizations.supportedLocales,
+          home: GlobalStoreWidget(
+            child: ChooseAccountPage())));
+
+      // global store gets loaded
+      await tester.pumpAndSettle();
+    }
+
+    List<Account> dummyAccounts(int count) {
+      return List.generate(count, (i) => eg.account(
+        id: i,
```

Comment:
Is this better here or moved to the `example_data.dart` file?

Reply:
Here is good. The `example_data.dart` file is good for things that are likely to be reused across a broad range of tests; but I think it'll be fairly uncommon that we want to test having a lot of accounts.